src/main/java/com/example/demo/controller/EditController.java

```diff
@@ -49,17 +47,16 @@ public String postUpdatePattern(@Validated Pattern pattern,
 			return getPattern(model);
 		}
 		
-		dateCalcMapper.updatePattern(pattern);
+		dateCalcService.updateCalcPattern(pattern);
 		
-		return "redirect:/pattern";
+		return "redirect:/date-calculation/edit";
 	}
 	
-	@PostMapping(value="/pattern", params = "delete")
-	public String postDeletePattern(@RequestParam("delete") String strNumber) {
+	@PostMapping(params = "delete")
+	public String postDeletePattern(@RequestParam("delete") String strId) {
```

Comment:
↓にしましょう
```suggestion
	public String postDeletePattern(@RequestParam("delete") String id) {
```

変数の接頭辞に型名をつけるのはハンガリアン記法というもので現在は推奨されていない書き方です！
`ハンガリアン記法`で調べてみてください！